src/main/steps/applicant1/enter-their-address/get.ts

```diff
@@ -0,0 +1,27 @@
+import autobind from 'autobind-decorator';
+import { Response } from 'express';
+
+import { State, YesOrNo } from '../../../app/case/definition';
+import { AppRequest } from '../../../app/controller/AppRequest';
+import { GetController } from '../../../app/controller/GetController';
+
+@autobind
+export default class EnterTheirAddressGetController extends GetController {
+  public async get(req: AppRequest, res: Response): Promise<void> {
+    if (
+      req.session.userCase.applicant2AddressPrivate === YesOrNo.YES &&
+      req.session.userCase.state === State.AwaitingApplicant1Response
+    ) {
+      req.session.userCase.applicant2Address1 = undefined;
```

Comment:
is this to hide applicant2 details from applicant1? shouldn't we just obfuscate that in the template rather than actually changing the data

Reply:
the template uses the session data so im not sure how we should do that.

Reply:
maybe a custom function which gets the userCase and we override it in the custom controller?